gateway/mw_request_size_limit_test.go

```diff
@@ -60,4 +66,31 @@ func TestRequestSizeLimit(t *testing.T) {
 			}...)
 		}
 	})
+
+	t.Run("should break the request, if content-length is missing", func(t *testing.T) {
+		// Golang's HTTP client automatically adds Content-Length to the request for POST, PUT and PATCH methods.
+		logger, _ := logrus.NewNullLogger()
+		spec := ts.Gw.BuildAndLoadAPI(func(spec *APISpec) {
+			UpdateAPIVersion(spec, "v1", func(v *apidef.VersionInfo) {
+				v.GlobalSizeLimit = 1024
+			})
+		})[0]
+		baseMid := &BaseMiddleware{
+			Spec:   spec,
+			logger: logger.WithContext(context.Background()),
+		}
+		reqSizeLimitMiddleware := &RequestSizeLimitMiddleware{baseMid}
+
+		for _, method := range []string{http.MethodPost, http.MethodPut, http.MethodPatch} {
+			// Content-Length is missing in this request.
+			body := bytes.NewBufferString(strings.Repeat("a", 3))
+			r, err := http.NewRequest(method, "/sample", body)
```

Comment:
I think you can use httptest.NewRequest here, linter reports issues
https://pkg.go.dev/net/http/httptest#NewRequest

Reply:
@titpetric the problem has addressed